src/Adapter/MSTest.TestAdapter/Execution/UnitTestRunner.cs

```diff
@@ -200,33 +226,57 @@ internal UnitTestResult[] RunSingleTest(TestMethod testMethod, IDictionary<strin
         return result;
     }
 
-    private void RunClassCleanupIfEndOfClass(TestMethodInfo testMethodInfo, TestMethod testMethod, UnitTestResult[] results)
+    private void TryToRunCleanups(ITestContext testContext, TestMethodInfo testMethodInfo, TestMethod testMethod, UnitTestResult[] results)
```

Comment:
I usually prefer to keep `TryXXX` prefix for method returning a bool (i.e. using the try pattern). If your idea is to suggest that it's not always running cleanup you could go with `RunCleanupsIfNeeded`.